test/internals/reply.test.js

```diff
@@ -82,6 +82,17 @@ test('within an instance', t => {
     reply.send({hello: 'world!'})
   })
 
+  fastify.register(function (instance, options, next) {
+    fastify.addHook('onSend', function (req, reply, payload, next) {
+      reply.header('x-onsend', 'yes')
```

Comment:
The header is set but there's no assert for this

Reply:
See my note in the PR. It would be more complete to check for it, it isn't possible.

Reply:
Use `inject`

Reply:
The test is a literal copy and past of the `redirect / - 4` test and modified to test the new functionality. All other tests in the file use `simple-get`. I'm not opposed to changing this test, but I will need an example of how to do it.

Reply:
I wasn't thinking. I have updated the test.